### PR TITLE
Add benefit selection to 1990e

### PR DESCRIPTION
**Pull request by cvalarida**
Connects to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/631

---
image

### CODE REVIEW

**Comment by jbalboni** [comment]
@cvalarida Needs a screenshot, too.

File: src/js/edu-benefits/1990e/config/form.js [hunk 1]
```diff
@@ -25,6 +39,31 @@ const formConfig = {
     benefitEligibility: {
       title: 'Benefit Eligibility',
       pages: {
+        benefitEligibility: {
+          path: 'benefit-eligibility',
+          title: 'Benefit Eligibility',
+          uiSchema: {
+            'view:benefitDescription': {
+              'ui:description': eligibilityDescription
```
Comment (jbalboni) [inline src/js/edu-benefits/1990e/config/form.js]: You could try putting this just on the uiSchema object:

```js
uiSchema: {
  'ui:description': eligibilityDescription,
  benefit: {
    ...
  }
}
```

**Comment by cvalarida** [comment]
Right! Updated with the screenshot. Thanks, @jbalboni.

File: src/js/edu-benefits/1990e/helpers.jsx [hunk 2]
```diff
@@ -8,3 +10,24 @@ export function transform(formConfig, form) {
...
+          <li>You can only get payments from 1 program at a time.</li>
+          <li>You can’t get more than 48 months of benefits under any combination of VA education programs.</li>
+        </ul>
+      </div>
+    </div>
+  );
+}
+
+export const benefitsLabels = {
+  chapter33: <p>Post-9/11 GI Bill (Chapter 33)<br/><a href="/education/gi-bill/post-9-11/" target="_blank">Learn more</a></p>,
```
Comment (jbalboni) [inline src/js/edu-benefits/1990e/helpers.jsx]: Are these the same as the 1995? Maybe we should put them in a helpers file in edu-benefits/utils that both forms could share?

**Review by jbalboni — COMMENTED** [review]
Looks good. I made two small comments. You should also add a test for this page.

**Comment by jbalboni** [comment]
@cvalarida You can do the schema update in this PR if you'd like. You'll have to merge from master first, since I had to fix something on the 1995 for it to work with the most recent schema change.

**Comment by cvalarida** [comment]
With master merged, the schema updated, benefits labels shared, and tests added, can you take another look, @jbalboni? Should be good, but I like the assurance of double-checking.

File: src/js/edu-benefits/utils/helpers.js [hunk 3]
```diff
@@ -14,6 +15,15 @@ export const chapterNames = {
   review: 'Review'
 };
 
+export const benefitsLabels = {
+  chapter33: <p>Post-9/11 GI Bill (Chapter 33)<br/><a href="/education/gi-bill/post-9-11/" target="_blank">Learn more</a></p>,
```
Comment (raquelromano-gov) [inline src/js/edu-benefits/utils/helpers.js]: I'm surprised these don't need to be surrounded by quotation marks, although it seems to have worked before. What happens if you add quotes?

File: src/js/edu-benefits/1990e/config/form.js (hunk 1 above)
Comment (jbalboni) [inline src/js/edu-benefits/1990e/config/form.js]: @cvalarida Did this not work?

File: src/js/edu-benefits/utils/helpers.js (hunk 3 above)
Comment (jbalboni) [inline src/js/edu-benefits/utils/helpers.js]: This file should have a jsx extension now.

File: test/edu-benefits/1990e/config/benefitSelection.unit.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+          schema={schema}
+          onSubmit={onSubmit}
+          data={{}}
+          uiSchema={uiSchema}/>
+    );
+    const formDOM = findDOMNode(form);
+    submitForm(form);
+    expect(Array.from(formDOM.querySelectorAll('.usa-input-error'))).to.be.empty;
+
+    submitForm(form);
```
Comment (jbalboni) [inline test/edu-benefits/1990e/config/benefitSelection.unit.spec.jsx]: I don't think you need to call this twice.

**Review by jbalboni — APPROVED** [review]
Couple comments, but looks good.

File: src/js/edu-benefits/utils/helpers.js (hunk 3 above)
Comment (cvalarida) [inline src/js/edu-benefits/utils/helpers.js]: I tried giving it `.jsx`, but it broke in the 1990 routes, so I changed the import statement there and it broke in two more places that...I couldn't even trace. Funky stuff was going on...I'll try it again, though.

Comment (cvalarida) [inline src/js/edu-benefits/utils/helpers.js]: Hmm. Now it works just fine. Guess I'll go with it.

Comment (cvalarida) [inline src/js/edu-benefits/utils/helpers.js]: Turns out I just need to restart `npm run watch` after files are renamed...which is odd, but that showed me the proper files that specifically referred to `helpers.js` (with the extension).

Comment (jbalboni) [inline src/js/edu-benefits/utils/helpers.js]: @raquelromano-gov The lack of quotes means that value is interpreted as jsx, which is compiled into function calls by Babel, like Chris mentioned.

By convention, we name files `.jsx` if they contain any jsx, but we're transpiling both `.js` and `.jsx` files.

**Pull request merged**